docs/authorization.md

```diff
@@ -0,0 +1,43 @@
+# Proposal(s), Session(s), and related entities
+
+Authorization is applied to all user facing resources in py-ISPyB and different permissions are available to grant users and staff access to entities related to the core of ISPyB. These include but not limited to:
+
+- Proposal
+- Protein, Crystal, BLSample, Shipping, LabContact
+- BLSession, DataCollectionGroup, DataCollection
+
+etc ...
+
+The authorization rules are applied in four ways:
+
+### Users
+
+- A user can access entities related to a Proposal and the DataCollection(s) in which they are a member of one or more Session(s) [linked via SessionHasPerson]. _This is an intrinsic permission and is the default behaviour if the user has no other permissions._
+- A user can access entities related to all Session(s) in a Proposal [linked via ProposalHasPerson]
+
+### Administrators
+
+- An administrator can view all Sessions on a Proposal for specific beamline(s) via a `BeamLineGroup` permission
+- An administrator can access all Sessions and Proposals via `all_proposals`
+
+## BeamLineGroups
+
+Beamline groups provide a way to grant access to all Proposals, Sessions and related entities to a set of staff members for a particular group of beamlines.
+
+For example:
+
+```json
+"beamLineGroups": [
+    {
+        "groupName": "BL0x",
+        "uiGroup": "mx",
+        "permission": "bl0_admin",
+        "beamlines": [
```

Comment:
because you can archive them too

Reply:
What is the use case for archiving a beamline in a group?

Reply:
Maybe that should be a UI setting then? I think it would belong more there than in authorization setting.

 We have this configuration object for the beamlines:
```ts
techniques = {
      SAXS: { beamlines: [{ name: 'BM29' }] },
      EM: { beamlines: [{ name: 'CM01' }] },
      MX: {
        beamlines: [
          { name: 'ID23-1', sampleChangerType: 'FlexHCDDual' },
          { name: 'ID23-2', sampleChangerType: 'FlexHCDUnipuckPlate' },
          { name: 'ID29', sampleChangerType: 'FlexHCDDual' },
          { name: 'ID30A-1', sampleChangerType: 'FlexHCDUnipuckPlate' },
          { name: 'ID30A-2', sampleChangerType: 'FlexHCDDual' },
          { name: 'ID30A-3', sampleChangerType: 'FlexHCDDual' },
          { name: 'ID30B', sampleChangerType: 'FlexHCDDual' },
          { name: 'BM30A', sampleChangerType: 'FlexHCDDual' },
        ],
      },
    }
```
This allows the UI to show only the beamlines that are associated with the selected technique. If we want to hide the decommissioned beamlines, we can just remove them from that list. If we want to have the option to show/hide decommissioned beamlines, we could add an extra field like `archive` or `active` or `decommissioned`.

Reply:
It deals with both authorisation and some information for the UI side, we dont want to split this information into two places (confusing the maintain). `beamLineGroup` is automatically exported to the UI via `/options/ui`, so all you need to do is add `sampleChangerType` to the schema, and you have the above information served from the backend automatically...

Reply:
Added `sampleChangerType` and `sampleChangerCapacity` fallback to options, and these are available to the ui